mailbox/noise_test.go

```diff
@@ -27,3 +33,132 @@ func TestSpake2Mask(t *testing.T) {
 	unmaskedPoint := ekeUnmask(maskedPoint, passHash[:])
 	require.True(t, unmaskedPoint.IsEqual(pub))
 }
+
+// TestXXHandshake tests that a client and server can successfully complete a
+// Noise_XX pattern handshake and then use the encrypted connection to exchange
+// messages afterwards.
+func TestXXHandshake(t *testing.T) {
+	// First, generate static keys for each party.
+	pk1, err := btcec.NewPrivateKey(btcec.S256())
+	require.NoError(t, err)
+
+	pk2, err := btcec.NewPrivateKey(btcec.S256())
+	require.NoError(t, err)
+
+	// Create a password that will be used to mask the first ephemeral key.
```

Comment:
Seems to have a lot of overlap with these tests? https://github.com/lightninglabs/lightning-node-connect/blob/master/mailbox/tcp_noise_test.go#L108

Reply:
I guess this is from a slightly higher layer of abstraction though, the gRPC level.

Reply:
> afaict, we dont use the code in tcp_noise_conn.go and tcp_noise_listener.go anywhere?

Correct, but we can use them when we eventually add the pure TCP version of LNC, for nodes that don't really need the NAT traversal component.